tests/Unit/Evolution/Systems/CurvedScalarWave/BoundaryCorrections/Test_UpwindPenalty.cpp

```diff
@@ -0,0 +1,310 @@
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Minkowski.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+
+namespace {
+template <size_t Dim>
+void test(const gsl::not_null<std::mt19937*> gen, const size_t num_pts) {
+  PUPable_reg(CurvedScalarWave::BoundaryCorrections::UpwindPenalty<Dim>);
+
+  TestHelpers::evolution::dg::test_boundary_correction_with_python<
+      CurvedScalarWave::System<Dim>>(
+      gen,
+      "Evolution.Systems.CurvedScalarWave.BoundaryCorrections.UpwindPenalty",
+      {{"dg_package_data_v_psi", "dg_package_data_v_zero",
+        "dg_package_data_v_plus", "dg_package_data_v_minus",
+        "dg_package_data_gamma2", "dg_package_data_interface_unit_normal",
+        "dg_package_data_char_speeds"}},
+      {{"dg_boundary_terms_pi", "dg_boundary_terms_phi",
+        "dg_boundary_terms_psi"}},
+      CurvedScalarWave::BoundaryCorrections::UpwindPenalty<Dim>{},
+      Mesh<Dim - 1>{num_pts, Spectral::Basis::Legendre,
+                    Spectral::Quadrature::Gauss},
+      {}, {});
+
+  const auto upwind_penalty = TestHelpers::test_creation<std::unique_ptr<
+      CurvedScalarWave::BoundaryCorrections::BoundaryCorrection<Dim>>>(
+      "UpwindPenalty:");
+
+  TestHelpers::evolution::dg::test_boundary_correction_with_python<
+      CurvedScalarWave::System<Dim>>(
+      gen,
+      "Evolution.Systems.CurvedScalarWave.BoundaryCorrections.UpwindPenalty",
+      {{"dg_package_data_v_psi", "dg_package_data_v_zero",
+        "dg_package_data_v_plus", "dg_package_data_v_minus",
+        "dg_package_data_gamma2", "dg_package_data_interface_unit_normal",
+        "dg_package_data_char_speeds"}},
+      {{"dg_boundary_terms_pi", "dg_boundary_terms_phi",
+        "dg_boundary_terms_psi"}},
+      dynamic_cast<
+          const CurvedScalarWave::BoundaryCorrections::UpwindPenalty<Dim>&>(
+          *upwind_penalty),
+      Mesh<Dim - 1>{num_pts, Spectral::Basis::Legendre,
+                    Spectral::Quadrature::Gauss},
+      {}, {});
+}
+
+template <size_t Dim>
+void test_flat_spacetime(const gsl::not_null<std::mt19937*> gen) {
+  INFO("test consistency of the curved-scalar-wave upwind flux")
+  std::uniform_real_distribution<> uniform_m11_dist(0.1, 1.);
+
+  const size_t num_pts = 3;
+  const DataVector value{1., 2., 3.};
+  const tnsr::I<DataVector, Dim> x{value};
+  const double t = 1.3;
+
+  gr::Solutions::Minkowski<Dim> minkowski{};
+  ScalarWave::BoundaryCorrections::UpwindPenalty<Dim> sw_flux_computer{};
+  CurvedScalarWave::BoundaryCorrections::UpwindPenalty<Dim> csw_flux_computer{};
+
+  // buffer storing variables for CSW
+  TempBuffer<tmpl::list<
```

Comment:
Hmm. I sort of disagree on this. I would prefer leaving it here as is, if you don't mind too much. 

The reduction to readability of the test is limited to the declaration of the buffer itself, after which we are simply extracting references to memory in it - which just replaces allocating memory. 

And as spectre adds more and more unit tests, why would we not care about the performance of unit tests?

Reply:
I think since the build time of unit tests is far longer than the runtime and this allocation is not in any kind of loop, the reduced memory allocation speed-up will not be noticeable at all.

Overall I just feel strongly that unit tests should be as explicit and clear as possible making them easy to debug and useful for understanding how the tested code works. But I dont want to impose my opinions here too strongly so I would be fine with approving it like this.

Reply:
That is fair. what also changed my mind is that this would be the only unit test that actually uses temp buffers, so I've removed their usage here as well.